shared/js/background/tab-manager.es6.js

```diff
@@ -36,6 +40,12 @@ class TabManager {
      * get({tabId: ###});
      */
     get (tabData) {
+        // Opaque 'tab' (e.g. ServiceWorker). Create the Tab Object, but don't
+        // store it for later lookup.
+        if (tabData.tabId === -1) {
```

Comment:
So normal tabs go through "create" and then can be fetched via "get", but SW tabs don't go through "create"?

Reply:
Yea, to be honest I ran the code through a debugger to figure out what happens and what needs to happen. [At the start of handleRequest, we call tabManager.get()](https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/e3ebc63/shared/js/background/redirect.es6.js#L89) to fetch the Tab Object for the request. For requests initiated by ServiceWorkers, there won't be a Tab Object and so we need to return a temporary opaque Tab Object if we want tracker blocking to work.

In other words there were three things stopping tracker blocking working for such requests:
1. We explicitly ignored requests with tab ID of -1
2. We had no Tab Object for the "tab" (because its not a tab)
3. When creating an temporary Tab Object, the site URL and tab URL was the same and so the request was always considered first-party